### PR TITLE
[CPDNPQ 1008] Add data filters to edge case page

### PR DESCRIPTION
**Pull request by javier-npq**
### Context

- Ticket: https://dfedigital.atlassian.net/browse/CPDNPQ-1008

### Changes proposed in this pull request

- added data filters for fundingEligibilityStatusCode
- added data filters for employmentType
- added specs for new data filters

### CODE REVIEW

File: app/controllers/admin/npq/applications/edge_cases_controller.rb [hunk 1]
```diff
@@ -13,9 +13,13 @@ class EdgeCasesController < Admin::BaseController
 
         def index
           query_string = params[:query]
+          funding_eligiblity_status_code = params[:FundingEligiblityStatusCode]
+          employment_type = params[:EmploymentType]
+          start_date = convert_date(params["start_date(3i)"], params["start_date(2i)"], params["start_date(1i)"])
```
Comment (shjohnson) [inline app/controllers/admin/npq/applications/edge_cases_controller.rb:18]: is there anyway to give more human readable param names? something like `start_date(day)`, `start_date(month)` etc

File: app/models/npq_application.rb [hunk 2]
```diff
@@ -50,6 +50,7 @@ class NPQApplication < ApplicationRecord
                                                                         marked_ineligible_by_policy
                                                                         marked_funded_by_policy])
                              }
+  scope :created_at_range, ->(start_date, end_date) { where(created_at: start_date..end_date) }
```
Comment (shjohnson) [inline app/models/npq_application.rb:53]: 👏

File: app/services/admin/npq_applications/edge_case_search.rb [hunk 3]
```diff
@@ -37,6 +45,30 @@ def search_conditions
     end
   end
 
+  def funding_eligiblity_status_code_conditions
```
Comment (shjohnson) [inline app/services/admin/npq_applications/edge_case_search.rb:48]: Was going to say all these methods are very similar and could be dried up but i think this is fine for just 3.

File: spec/services/admin/npq_applications/edge_case_search_spec.rb [hunk 4]
```diff
@@ -102,5 +102,48 @@
         expect(subject.call).not_to include(application_2)
       end
     end
+
```
[the remark this answers is not in the document]
Comment (shjohnson) [reply spec/services/admin/npq_applications/edge_case_search_spec.rb:105]: and not the sub query of the different statuses or dates?

**Comment by shjohnson** [comment]
Can wee add some randomised seed data with different dates for the edge cases mainly for the review app when it gets built. possibly as well other bits of data.

File: app/controllers/admin/npq/applications/edge_cases_controller.rb (hunk 1 above)
Comment (javier-npq) [reply app/controllers/admin/npq/applications/edge_cases_controller.rb:20]: Yeah that's fiddly. Left a comment for now, Went through the [documentation for this component here ](https://www.rubydoc.info/gems/govuk_design_system_formbuilder/GOVUKDesignSystemFormBuilder/Builder#govuk_date_field-instance_method)and couldn't find anything out of the box that we can use.

File: app/services/admin/npq_applications/edge_case_search.rb (hunk 3 above)
Comment (javier-npq) [reply app/services/admin/npq_applications/edge_case_search.rb:48]: I did try to do it like this in the beginning which was a much cleaner way image
But didn't get it running due to the way the flow is set up. Kept on getting the following error.
image
I can move them to a separate helper/concern class and call it from there but it's just three methods so i left them here.

File: spec/services/admin/npq_applications/edge_case_search_spec.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (javier-npq) [reply spec/services/admin/npq_applications/edge_case_search_spec.rb:107]: Yeah, they are now.

**Pull request merged**